server/routes/users.js

```diff
@@ -93,598 +93,596 @@ router.get('/login', redirectProfile, (req, res) => {
-			req.session.user = dbUser;
-			res.locals.user = dbUser;
-			return res.status(200).send({
-				url: '/users/profile',
-			});
-		}
-
-		try {
-			const newUser = await UserRepository.createNewUser({
-				fName,
-				lName: lName || 'LastnameUnset',
-				email: email.toLowerCase(),
-				password: createDefaultPassword(),
-				verificationHash: createEmailVerificationHash(),
-				userRole: 'donor',
-				loginMode: 'Facebook',
-				emailVerified: true,
-			});
-
-			req.session.user = newUser;
-			res.locals.user = newUser;
-			return res.status(200).send({
-				url: '/users/profile',
-			});
-		} catch (error) {
-			return handleError(res, 400, 'Error during login!\nTry again in a few minutes!');
-		}
-	}
-
-	return handleError(res, 400, 'Error during login!\nTry again in a few minutes!');
+  const { userName, email } = req.body;
+
+  if (userName && email) {
+    const [fName, lName] = userName.split(' ');
+
+    const dbUser = await UserRepository.getUserByEmail(email.toLowerCase());
+
+    if (dbUser) {
+      req.session.user = dbUser;
+      res.locals.user = dbUser;
+      return res.status(200).send({
+        url: '/users/profile',
+      });
+    }
+
+    try {
+      const newUser = await UserRepository.createNewUser({
+        fName,
+        lName: lName || 'LastnameUnset',
+        email: email.toLowerCase(),
+        password: createDefaultPassword(),
+        verificationHash: createEmailVerificationHash(),
+        userRole: 'donor',
+        loginMode: 'Facebook',
+        emailVerified: true,
+      });
+
+      req.session.user = newUser;
+      res.locals.user = newUser;
+      return res.status(200).send({
+        url: '/users/profile',
+      });
+    } catch (error) {
+      return handleError(res, 400, 'Error during login!\nTry again in a few minutes!');
+    }
+  }
+
+  return handleError(res, 400, 'Error during login!\nTry again in a few minutes!');
 });
 
 // @desc    Render login.html
 // @route   POST '/users/login'
 // @access  Public
 // @tested 	Not yet
 router.post(
-	'/login',
-	limiter,
-	loginValidationRules(),
-	validate,
-	redirectProfile,
-	async (req, res) => {
-		const { email, password } = req.body;
-
-		const user = await UserRepository.getUserByEmail(email.toLowerCase());
-		if (user) {
-			if (await bcrypt.compare(password, user.password)) {
-				req.session.user = user;
-				res.locals.user = user;
-				return res.status(200).send({ success: true, url: '/users/profile' });
-			}
-		}
-		handleError(res, 403, 'Username and/or password incorrect');
-	},
+  '/login',
+  limiter,
+  loginValidationRules(),
+  validate,
+  redirectProfile,
+  async (req, res) => {
+    const { email, password } = req.body;
+
+    const user = await UserRepository.getUserByEmail(email.toLowerCase());
+    if (user) {
+      if (await bcrypt.compare(password, user.password)) {
+        req.session.user = user;
+        res.locals.user = user;
+        return res.status(200).send({ success: true, url: "/users/profile" })
```

Comment:
wrong "?

Reply:
old server, that's from merging the master